### DISCUSSION TITLE
Allow Enterprise (Cloud) Administrators to perform CRUD operations on Audit Stream Programatically

### DISCUSSION THREAD

**Discussion by elliott-weston-cko** [community/community#13746] [Enterprise]
Hello GitHub!,

My company has been exploring the usage of the GitHub Audit Streaming Feature that's available for Enterprise Cloud users: https://docs.github.com/en/enterprise-cloud@latest/admin/user-management/managing-organizations-in-your-enterprise/streaming-the-audit-logs-for-organizations-in-your-enterprise-account

The majority of the storage providers that are supported use static credentials to provide GitHub with access to push the audit events to the storage provider.
From a security perspective, it is best practice to frequently rotate static credentials to reduce the risk in the event the credentials are leaked.

This is currently difficult to do, as there is no API to allow Enterprise Administrators (or Service Accounts with Enterprise Administrator Access) to programmatically rotate these keys.

As a prospective user of this very useful feature, I'd like a way to programmatically rotate these static credentials

**Comment by boylejj** [comment]
@elliott-weston-cko I am the new product manager for the GitHub audit log. We have recently begun exploring the use of OpenID connect to obfuscate the need for long lived tokens when configuring an audit log stream. See our [public roadmap tile](https://github.com/github/roadmap/issues/499). Would something like this address your concern with rotating static credentials?
